pytensor/link/mlx/linker.py

```diff
@@ -0,0 +1,94 @@
+from pytensor.link.basic import JITLinker
+
+
+class MLXLinker(JITLinker):
+    """A `Linker` that JIT-compiles NumPy-based operations using Apple's MLX."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.gen_functors = []
+
+    def fgraph_convert(self, fgraph, **kwargs):
+        """Convert a PyTensor FunctionGraph to an MLX-compatible function.
+
+        Parameters
+        ----------
+        fgraph : FunctionGraph
+            The function graph to convert
+
+        Returns
+        -------
+        callable
+            An MLX-compatible function
+        """
+        from pytensor.link.mlx.dispatch import mlx_funcify
+
+        return mlx_funcify(
+            fgraph,
+            **kwargs,
+        )
+
+    def jit_compile(self, fn):
+        import mlx.core as mx
+
+        from pytensor.link.mlx.dispatch import mlx_typify
+
+        class wrapper:
```

Comment:
This whole wrapper may not be needed this was stuff t appease torch compile machinery. Try to just have a simple callable like the other JITLinkers

Reply:
Doesnt support ndarray so there needs to be a conversion. Would that not happen here?